dev/devicelab/bin/run.dart

```diff
@@ -73,9 +75,27 @@ final ArgParser _argParser = new ArgParser()
     abbr: 't',
     allowMultiple: true,
     splitCommas: true,
-    help: 'Name of the task to run. This option may be repeated to '
-    'specify multiple tasks. A task selected by name does not have to be '
-    'defined in manifest.yaml. It only needs a Dart executable in bin/tasks.',
+    help: 'Either:\n'
+        ' - the name of a task defined in manifest.yaml. Example: complex_layout__start_up.\n'
+        ' - the path to a Dart file corresponding to a task, which resides in bin/tasks. Example: bin/tasks/complex_layout__start_up.dart.\n'
+        '\n'
+        'This option may be repeated to specify multiple tasks.',
+    callback: (List<String> value) {
+      for (String nameOrPath in value) {
+        List<String> fragments = path.split(nameOrPath);
+        bool isDartFile = fragments.last.endsWith('.dart');
+
+        if (fragments.length == 1 && !isDartFile) {
+          // Not a path
+          _taskNames.add(nameOrPath);
+        } else if (!isDartFile || fragments.length != 3 || !_listsEqual(<String>['bin', 'tasks'], fragments.take(2).toList())) {
+          // Unsupported executable location
+          throw new FormatException('Invalid value for option -t (--task): $nameOrPath');
```

Comment:
Either's fine by me. Two options means you now have to decide which one takes precedence, or catch both being set, etc. So it's not necessarily a win.